projects/plugins/jetpack/modules/publicize/publicize-jetpack.php

```diff
@@ -141,7 +201,7 @@ function get_connections( $service_name, $_blog_id = false, $_user_id = false )
 		if ( ! empty( $connections ) && is_array( $connections ) ) {
 			if ( ! empty( $connections[ $service_name ] ) ) {
 				foreach ( $connections[ $service_name ] as $id => $connection ) {
-					if ( 0 == $connection['connection_data']['user_id'] || $_user_id == $connection['connection_data']['user_id'] ) {
+					if ( 0 === (int) $connection['connection_data']['user_id'] || $_user_id === (int) $connection['connection_data']['user_id'] ) {
```

Comment:
Feel free to ignore this suggestion, but I was thinking if the following would be worth doing?
 
`if ( empty ( $connection['connection_data']['user_id'] )  || $_user_id === (int) $connection['connection_data']['user_id'] )` ?

or 

write a function called is_global_connection like this:

```
public function is_global_connection(  $connection ) {

   return empty(  $connection['connection_data']['user_id']  );
}
```

and then do 
`if ( self::is_global_connection( $connection) || $_user_id === (int) $connection['connection_data']['user_id'] ) `